graphql/executor/executor.go

```diff
@@ -227,6 +227,8 @@ func (e *Executor) parseQuery(
 		validator.RemoveRule("FieldsOnCorrectType")
 
 		rule := rules.FieldsOnCorrectTypeRuleWithoutSuggestions
+		// remove the rule added when it was last executed
+		validator.RemoveRule(rule.Name)
```

Comment:
If gqlparser has a function like Contains, it might be better to use it and add a conditional `if` check to determine whether the rule is already duplicated.

However, in terms of computational cost, the current implementation should not cause any issues. Therefore, I think it would be sufficient to leave this as a future TODO for now.